runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/testing/TestDataflowRunner.java

```diff
@@ -77,9 +77,11 @@
   public static TestDataflowRunner fromOptions(
       PipelineOptions options) {
     TestDataflowPipelineOptions dataflowOptions = options.as(TestDataflowPipelineOptions.class);
-    dataflowOptions.setStagingLocation(Joiner.on("/").join(
+    String tempLocation = Joiner.on("/").join(
         new String[]{dataflowOptions.getTempRoot(),
-          dataflowOptions.getJobName(), "output", "results"}));
+        dataflowOptions.getJobName(), "output", "results"});
+    dataflowOptions.setTempLocation(tempLocation);
+    dataflowOptions.setStagingLocation(tempLocation);
```

Comment:
I chatted with @jasonkuster offline.
Jenkins is not the place to set stagingLocation, since stagingLocation is a dataflow pipeline options.
tempLocation might be able to replace tempRoot(), but this is out of the scope of this PR.